gettsim/pre_processing/piecewise_functions.py

```diff
@@ -28,27 +28,42 @@ def piecewise_polynomial(
         The value of `x` under the piecewise function.
 
     """
+    # If no individual is transferred, we return an empty series
+    if x.empty:
+        return x
+    # # Check if missing data is given
+    # if x.isnull().values.any():
+    #     raise ValueError(f"There is missing data in {x.name}.")
 
     num_intervals = len(thresholds) - 1
     degree_polynomial = rates.shape[0]
-    # If now individual is transferred, we return an empty series
-    if x.empty:
-        return x
     # Check in which interval each individual is. The thresholds are not exclusive on
     # the right side!
-    binned = pd.cut(x, bins=thresholds, right=False, labels=range(num_intervals))
+    binned = pd.cut(
+        x,
+        bins=thresholds,
+        right=False,
+        include_lowest=True,
+        labels=range(num_intervals),
+    )
     # Create series with last threshold for each individual
     thresholds_individual = binned.replace(dict(enumerate(thresholds[:-1])))
     # Increment for each individual in the corresponding interval
     increment_to_calc = x - thresholds_individual
 
+    # Check if any value is in the lowest interval.
+    if 0 in binned.values:
```

Comment:
Must not happen anymore as we check that we cover everything in (-inf, inf) while setting things up.

Reply:
Still need to check if one value is in the interval starting at -inf. If we do not check the increment of the value over the lower threshold is value - (-inf) which yields nans!

Reply:
Please put into the issue to be created as per below. There should be a more elegant and clear way for this whole block of code, but this is not the right time to tackle it.

Reply:
Or `.array`, just used that one.